sycamore/evaluation/datasets.py

```diff
@@ -1,30 +1,23 @@
-from typing import Any, Union
+from typing import Union, Optional, Callable
 
 from datasets import IterableDataset
 from ray.data import Dataset, from_huggingface
 
-from sycamore.evaluation import EvaluationDataPoint
-from sycamore.scans import MaterializedScan
 from sycamore import DocSet, Context
+from sycamore.scans import MaterializedScan
 
 
 class HuggingFaceScan(MaterializedScan):
-    def __init__(self, dataset: Union[Dataset, IterableDataset], field_mapping: dict[str, str], **resource_args):
+    def __init__(
+        self, dataset: Union[Dataset, IterableDataset], doc_extractor: Optional[Callable] = None, **resource_args
```

Comment:
Can I get a type for doc_extractors. Also what happens if its none?

Reply:
I'll change it to required for now, and specify types for doc_extractor (just a function)